packages/golden_toolkit/lib/src/testing_tools.dart

```diff
@@ -128,29 +129,56 @@ void testGoldens(
   });
 }
 
-/// This [screenMatchesGolden] is wrapper on top of [matchesGoldenFile]
+/// A function that wraps [matchesGoldenFile] with some extra functionality. The function finds the first widget
+/// in the tree if [finder] is not specified. Furthermore a [fileNameFactory] can be used in combination with a [name]
+/// to specify a custom path and name for the golden file. In addition to that the function makes sure all images are
+/// available before
 ///
-/// [goldenFileName] is a file name output, must NOT include extension like .png
+/// [name] is the name of the golden test and must NOT include extension like .png.
+///
+/// [finder] is an optional finder, which can be used to target a specific widget to use for the test. If not specified
+/// the first widget in the tree is used
 ///
 /// [autoHeight] tries to find the optimal height for the output surface. If there is a vertical scrollable this
 /// ensures the whole scrollable is shown. If the targeted render box is smaller then the current height, this will
 /// shrink down the output height to match the render boxes height.
 ///
-/// [finder] optional finder
-///
 /// [customPump] optional pump function, see [CustomPump] documentation
 ///
 /// [skip] by setting to true will skip the golden file assertion. This may be necessary if your development platform is not the same as your CI platform
 Future<void> screenMatchesGolden(
   WidgetTester tester,
-  String goldenFileName, {
+  String name, {
+  bool autoHeight,
+  Finder finder,
+  CustomPump customPump,
+  bool skip,
+}) {
+  return compareWithGolden(
+    tester,
+    name,
+    autoHeight: autoHeight,
+    finder: finder,
+    customPump: customPump,
+    skip: skip,
+    device: null,
+    fileNameFactory: (String name, Device device) => GoldenToolkit.configuration.fileNameFactory(name),
```

Comment:
ah, I see this usage. I was thinking this could instead be GoldenToolkit.configuration.deviceFileNameFactory, then we could consolidate down to the one override and gracefully handle a null device. Otherwise, consumers will likely have to always supply both funcs. What do you think?

Reply:
I think it makes sense to have two different factories because they serve different purposes.
One is used when working with `screenMatchesGolden` and the other one when working with `multiScreenGolden`

Reply:
I can live with that for now. I suspect these two APIs will be consolidated down in the future, but we can wait to consolidate the factories until that point.